File: backend/src/main/java/com/go_exchange_easier/backend/dto/universityReview/GetUniversityReviewResponse.java
package com.go_exchange_easier.backend.dto.universityReview;

import java.time.Instant;
import java.util.List;

public record GetUniversityReviewResponse(

        Integer id,
        AuthorDto author,
        UniversityDto university,
        Short starRating,
        String textContent,
        Instant createdAt,
        List<ReactionDetailDto> reactions

) {

    public record AuthorDto(

            Integer id,
            String nick

    ) {}

    public record UniversityDto(

            Short id,
            String englishName,
            String nativeName

    ) {}

    public record ReactionDetailDto(

            Short typeId,
            String name,
            Short count,
            Boolean isSet

    ) {}

}

File: frontend/src/dtos/university-review/GetUniversityReviewResponse.ts
import type { ReactionName } from "../../types/ReactionName";

export type GetUniversityReviewResponse = {
  id: number;
  author: {
    id: number;
    nick: string;
  };
  university: {
    id: number;
    englishName?: string;
    nativeName: string;
  };
  starRating: number;
  textContent: string;
  createdAt: string;
  reactions: {
    typeId: number;
    name: ReactionName;
    count: number;
    isSet: boolean;
  }[]
}
